plonky2x/core/src/frontend/vars/stream.rs

```diff
@@ -50,6 +50,9 @@ impl<L: PlonkParameters<D>, const D: usize> OutputVariableStream<L, D> {
     }
     pub fn read<V: CircuitVariable>(&self, builder: &mut CircuitBuilder<L, D>) -> V {
         let variables = self.read_exact(builder, V::nb_elements());
+
+        // Reads from stream don't do any validity checks on the circuit variable.  It is the
+        // stream reader's responsibility to do so.
         V::from_variables_unsafe(&variables)
```

Comment:
Isn't this kind of bad? All places where we use streams, I think that we should do `V::from_variables` because when we're reading in `bytes` and stuff like that, isn't it good to have it be range-checked?